src/commands/openFromUrl.ts

```diff
@@ -0,0 +1,31 @@
+import * as fs from "fs";
+import * as vscode from "vscode";
+
+import { downloadAndExtract } from "./getAddon";
+import { getRootFolderPath } from "../utils/reviewRootDir";
+
+// handles urls of the form /review/<guid>/<version>
+export async function handleReviewUrl(guid: string, version: string) {
+  const rootPath = await getRootFolderPath();
+  const addonVersionManifestPath = `${rootPath}/${guid}/${version}/manifest.json`;
+  try {
+    await fs.promises.stat(addonVersionManifestPath);
+  } catch (error) {
+    await downloadAndExtract(guid, version);
+  }
+
+  const addonManifestUri = vscode.Uri.file(addonVersionManifestPath);
+  await vscode.window.showTextDocument(addonManifestUri);
+}
+
+// handles vscode://mozilla.assay/... urls
+export async function handleUri(uri: vscode.Uri) {
+  const { path } = uri;
+  const action = path.split("/")[1];
```

Comment:
you don't need to `.split` 3 times - something like `const [action, guid, version ...] = path.split("/")`  will do it

Reply:
I think it's fine to have it that way at the moment where we only support one action.  But if you _did_ want to future proof it then
```
const [action, ...restOfPath] = path.split("/");
...
const [guid, version] = restOfPath
```